### PR TITLE
[scaling] feat: make a log exporter for runtime

### PR DESCRIPTION
**Pull request by chesedo**
## Description of change
Move the runtime logging to a tracing layer and remove the stream subscriber from runtime

## How has this been tested? (if applicable)
Using the local runner

### CODE REVIEW

File: runtime/src/next/mod.rs [hunk 1]
```diff
@@ -288,8 +255,8 @@ impl Router {
         tokio::task::spawn_blocking(move || {
             let mut iter = logs_stream.bytes().filter_map(Result::ok);
 
-            while let Some(log) = Log::from_bytes(&mut iter) {
-                logs_tx.blocking_send(Ok(log.into())).expect("to send log");
+            while let Some(_log) = Log::from_bytes(&mut iter) {
+                // TODO: send to the log server here
```
Comment (oddgrd) [inline runtime/src/next/mod.rs:259]: Are you thinking we should do this in a follow-up?

**Review by oddgrd — APPROVED** [review]
Cool! LGTM, just a couple of questions, also clippy in CI wants a word :smile:

File: runtime/src/alpha/mod.rs [hunk 2]
```diff
@@ -215,8 +223,9 @@ where
         );
         trace!("got factory");
 
-        let logs_tx = self.logs_tx.clone();
-        let logger = Logger::new(logs_tx);
+        // TODO: make an argument when starting runtime
```
[the remark this answers is not in the document]
Comment (chesedo) [reply runtime/src/alpha/mod.rs:226]: :man_facepalming:

File: runtime/src/next/mod.rs (hunk 1 above)
Comment (chesedo) [reply runtime/src/next/mod.rs:259]: Yip, the wasm needs some other fixes first :sweat_smile:

File: cargo-shuttle/src/logger_server.rs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            .map(|u| u as u32),
+        target: fields
+            .remove("target")
+            .and_then(|v| v.as_str().map(|s| s.to_string()))
+            .unwrap_or_default(),
+        fields: serde_json::to_vec(&serde_json::Value::Object(fields)).unwrap_or_default(),
+    })
+}
+
+fn from_severity_number_to_level(severity_number: SeverityNumber) -> Level {
```
[the remark this answers is not in the document]
Comment (chesedo) [reply cargo-shuttle/src/logger_server.rs:102]: Hmm.... I might have a way to share them better...

**Review by iulianbarbu — COMMENTED** [review]
Just left one small comment. Other than this, we have to see what needs resolving from the existing comments from @oddgrd .

File: runtime/src/logger.rs [hunk 4]
```diff
@@ -1,113 +1,202 @@
...
+}
+
+/// Recorder to send logs over OTLP
+pub struct OtlpRecorder {
+    tx: mpsc::UnboundedSender<ScopeLogs>,
+    deployment_id: String,
+}
+
+impl OtlpRecorder {
+    /// Send deployment logs to `destination`. Also mark all logs as being generated by the `service_name`
```
[the remark this answers is not in the document]
Comment (chesedo) [reply runtime/src/logger.rs:30]: Copy + paste error :see_no_evil:

**Review by iulianbarbu — APPROVED** [review]
LGTM!

**Pull request merged**